File: src/Finance.java
class Java {

  public static void main(String[] args) {
    
  }
}

/**
 * Calculate compound interest of an investment and return the amount after specified time period
 */
public double compoundInterest(double principal, double monthlyRate, int numOfMonths) {
    return principal * Math.pow((1 + monthlyRate), numOfMonths);
}

// Calculate compound interest of an investment with additions every month and return the amount after specified time period
float calculateCompoundInterest(float principal, float interest, int years, float monthlyAddition){
    float compoundInterest = 0;
    for(int i = 1; i <= years; i++){
        
        compoundInterest += principal * (interest/100);
    principal += principal * (interest/100);
        
        for(int j = 1; j <= 12; j++){
            compoundInterest += monthlyAddition * (interest/100);
      principal += monthlyAddition * (interest/100);
        }
    }
    return compoundInterest;
}

calculateCompoundInterest(10000, 6, 5, 500);

void compound_interest(double p, double r, int n, int t) {
  double a = p * Math.pow(1 + (r / n), n * t);
  System.out.println("Amount = "+ a);
}
  
compound_interest(20000, .08, 12, 2);

// Calculate compound interest of an investment with additions every month and return the amount after a specified time period.
double compound_interest(double principal, double interest, int time, int freq) {
  double amount = principal * Math.pow(1 + (interest / freq), freq * time);
  return amount;
}

File: JavaScript/ScriptFiles/finance.js
#!/usr/bin/env node

// Calculate compound interest with monthly contributions
function CompoundInterest(principal, interestRate, contribution, years) {
    let balance = principal
    const months = years * 12
    for (let i = 1; i <= months; i++) {
        balance = balance + balance * interestRate / 12 + contribution
    }
    return balance
}
// Monthly contributions
function compound_interest(principal, annual_rate, monthly_rate, years) {
  
  let total = principal;
  
  for (let i = 0; i < years; i++) {
    total = total * (1 + annual_rate);
    total = total + monthly_rate;
  }
  
  return total;
}

function compoundInterest(principal, interest, years, contributions) {
  var amount = principal;
  for (var i = 0; i < years; i++) {
    amount = amount * (1 + interest / 12) + contributions;
  }
  return amount;
}

function compoundInterest(principal, interestRate, monthlyContribution, years){
    //calculate compound interest with monthly contributions
    const numberOfMonths = years * 12;
    let total = 0;
    let interest = 0;
    let newInterest = 0;
    let newPrincipal = 0;
    for(let i = 1; i <= numberOfMonths; i++){
        newPrincipal = total + monthlyContribution;
        interest = (newPrincipal * interestRate) / 12;
        newInterest = interest + newPrincipal;
        total = newInterest;
    }
    return total.toFixed(2);
}

compoundInterest(10000, 0.06, 1000, 12);

// Calculate compound interest of an investment with additions every month and return the amount after a specified time period.
function compoundInterest(p, r, m, t) {
    let ci = p * Math.pow((1 + (r / 12)), (12 * t));
    ci += m * ((Math.pow((1 + (r / 12)), (12 * t)) - 1) / (r / 12));
    return ci;
}
compoundInterest(1000, 0.05, 10, 1);


